ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java

```diff
@@ -188,6 +191,17 @@ public void authorize(Table table, Privilege[] readRequiredPriv, Privilege[] wri
       checkDeletePermission(path, getConf(), authenticator.getUserName());
     }
 
+    try {
+      String user = SecurityUtils.getUGI().getShortUserName();
```

Comment:
Can we move it before checkDeletePermission?

Reply:
if required will be done alter